src/Raven.Server/Documents/Indexes/MapReduce/Auto/AutoMapReduceIndex.cs

```diff
@@ -80,7 +80,31 @@ public override IIndexedItemEnumerator GetMapEnumerator(IEnumerable<IndexItem> i
 
         public override void Update(IndexDefinitionBaseServerSide definition, IndexingConfiguration configuration)
         {
-            SetPriority(definition.Priority);
+            var startIndex = false;
+            if (definition.ClusterState?.LastStateIndex > (Definition.ClusterState?.LastStateIndex ?? -1))
```

Comment:
copy/paste, probably would be better to extract this to some helper or a base class?

Reply:
done